### DISCUSSION TITLE
Issue with free space on Volume group when an OS is installed with custom installation in file system creation

### DISCUSSION THREAD

**Discussion by solaianandh** [oracle/oracle-linux#98] [General Oracle Linux questions]
Hi Team,

we have been seeing an issue/ may be an bug with OS installation approached with manual file system creation , the free space in VG is not showing post OS is booted up. eg:- an 100Gb disk is used for os creation, consider 28 gb is left in vg, post os booted up we dont see any free space left at all in vgs, where as one has to manually go fdisk -l and clear the space on 8e for this to be visible. 

doesnt it considered to be an major risk - just curious to know as we have seen teams not having much clarity has deleted the whole file system while tending to this free space.

please advise how we can fix it / stream line. 

looking for expertise feedback , TIA. cheers!

**Comment by AmedeeBulle** [comment]
The size of the VG will always be the size of the partition created during the manual partitioning.
If you don't allocate the remaining size of the disk to the partition, there will be unused space after the last partition.

Note that it is a common situation for cloud images, allowing the user to provision a bigger disk than the image.
To minimize human error while resizing the last partition, the command `growpart` (from the `cloud-utils-growpart` package) will do that for you.

**Comment by solaianandh** [comment]
Hey , thanks for the support in sharing your thoughts.

this doesnt limit with cloud images it happened to all physical servers and also vms and indeed its extending with cloud images as well. 

we have teams faced issues by human errors manually attempting to delete and ended with issues 

looking for your feedback , cheers!

**Comment by AmedeeBulle** [comment]
Well as I said, you basically have 2 options:

First, configure your system properly at install / partitioning time; that is: allocate all the (free) space to your volume group. That way you won't have to change your partitioning after the facts.
In case you aren't aware of this, by default the VG will be sized for the LVs you create in it, but you can give any size -- e.g. use all remaining space:
![Screenshot 2023-08-10 at 10 16 16](https://github.com/oracle/oracle-linux/assets/4170863/9e74c18d-4fb5-409e-bd3a-4ba4c7929b61)

Alternatively, if your system hasn't been properly partitioned, consider using `growpart`, it also works on bare metal and will minimize human error.

**Comment by solaianandh** [comment]
HI,

let me share artifacts, at the earliest. 

cheers!
